pkg/storage/snapshot/restore.go

```diff
@@ -1007,6 +1141,26 @@ func (ctrl *VMRestoreController) deleteObsoleteDataVolumes(vmRestore *snapshotv1
 		}
 	}
 
+	// Garbage-collect old backend PVC if needed.
+	// Target should always exist at this point, just nil check for safety.
+	if target.Exists() && backendstorage.IsBackendStorageNeededForVM(target.VirtualMachine()) {
+		// Garbage-collect original backend PVC
+		pvcs, err := ctrl.Client.CoreV1().PersistentVolumeClaims(vmRestore.Namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", restoreCleanupBackendPVCLabel, getCleanupLabelValue(vmRestore)),
+		})
+		if err != nil {
+			return err
+		}
+		if len(pvcs.Items) == 0 {
```

Comment:
Why not just iterate over items in the list and delete?

Reply:
ACK